### PR TITLE
REFACTOR: healing aura from pilons and Rod of Asclepus

### PR DESCRIPTION
**Pull request by ErdGinalD**
## What Does This PR Do
A healing aura refactor that will be needed in the future when porting a space dragon as an antagonist from TG. 

Now the heal is not tied to a long cd with a large recoil, but to second-by-second regeneration, which feels kind of faster, but in fact the speed of healing has not changed in any way.  The source of the healing aura is also now indicated, which will help the bearers of the Rod of Asclepius. 

At the moment, only cult pylons and Rod of Asclepius have the healing component, but this will change in the future.

## Why It's Good For The Game
I want more dragons, this refactor is good to peoples too.

## Images of changes
Heal from cult pylons(only for cultists)
image

Holder of ROA now indicate himself and pretty obvious to see.
image

## Testing
A lot of copypasta from previous build and adaptations. Healed 300 HP Ian, healed cultist, regular men, as bearer of rod itself.

## Changelog
:cl:
tweak: Cult pylons and Rod of Asclepius now heals second-by-second instead of huge intervals. Amount of heal greatly redused to match new regeneration speed.
tweak: Holder of Rod of Acklepius now alert everyone when someone comes into heal radius.
/:cl:

### CODE REVIEW

File: code/datums/components/aura_healing.dm [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	var/list/organ_healing = null
+
+	/// Shows that aura heals fractures
+	var/heal_fractures = FALSE
+
+	/// Shows that aura heals internal bleedings
+	var/heal_internal_bleedings = FALSE
+
+	/// Showa that aura heals critical burns
+
```
Comment (DGamerL) [inline code/datums/components/aura_healing.dm:49]:
```suggestion
	/// Shows that aura heals critical burns
```

File: code/datums/components/aura_healing.dm [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	/// Map of organ (such as ORGAN_SLOT_BRAIN) to damage heal over a second
+	var/list/organ_healing = null
+
+	/// Shows that aura heals fractures
+	var/heal_fractures = FALSE
+
+	/// Shows that aura heals internal bleedings
+	var/heal_internal_bleedings = FALSE
+
+	/// Showa that aura heals critical burns
```
Comment (DGamerL) [inline code/datums/components/aura_healing.dm:48]: I don't think the wording is correct, it should probably be something like `Should this aura heal [x]`

File: code/datums/components/aura_healing.dm [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+	src.suffocation_heal = suffocation_heal
+	src.stamina_heal = stamina_heal
+	src.clone_heal = clone_heal
+	src.blood_heal = blood_heal
+	src.organ_healing = organ_healing
+	src.heal_fractures = heal_fractures
+	src.heal_internal_bleedings = heal_internal_bleedings
+	src.simple_heal = simple_heal
+	src.limit_to_trait = limit_to_trait
+	src.healing_color = healing_color
```
Comment (DGamerL) [inline code/datums/components/aura_healing.dm:102]: I'd prefer it if you made the arguments start with an underscore, so we don't need the `src.`

File: code/datums/components/aura_healing.dm [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+/datum/component/aura_healing/Destroy(force, silent)
+	STOP_PROCESSING(SSaura_healing, src)
+	var/alert_category = "aura_healing_[\ref(src)]"
```
Comment (DGamerL) [inline code/datums/components/aura_healing.dm:106]: Can you use UIDs instead of refs here?

File: code/game/gamemodes/cult/cult_structures.dm [hunk 5]
```diff
@@ -257,52 +268,36 @@ GLOBAL_LIST_INIT(blacklisted_pylon_turfs, typecacheof(list(
...
-				// Are we in space or something? No cult turfs or
-				// convertable turfs?
-				last_corrupt = world.time + corrupt_delay * 2
+			validturfs |= T
+
+	var/turf/T = safepick(validturfs)
+	if(T)
+		if(istype(T, /turf/simulated/floor))
+			T.ChangeTurf(/turf/simulated/floor/engine/cult)
+		if(istype(T, /turf/simulated/wall))
```
Comment (Sirryan2002) [inline code/game/gamemodes/cult/cult_structures.dm:290]:
```suggestion
		if(iswallturf(T))
```

File: code/game/gamemodes/cult/cult_structures.dm [hunk 6]
```diff
@@ -257,52 +268,36 @@ GLOBAL_LIST_INIT(blacklisted_pylon_turfs, typecacheof(list(
...
-				new /obj/effect/temp_visual/cult/turf/open/floor(F)
-			else
-				// Are we in space or something? No cult turfs or
-				// convertable turfs?
-				last_corrupt = world.time + corrupt_delay * 2
+			validturfs |= T
+
+	var/turf/T = safepick(validturfs)
+	if(T)
+		if(istype(T, /turf/simulated/floor))
```
Comment (Sirryan2002) [inline code/game/gamemodes/cult/cult_structures.dm:288]:
```suggestion
		if(isfloorturf(T))
```

File: code/game/gamemodes/cult/cult_structures.dm [hunk 7]
```diff
@@ -257,52 +268,36 @@ GLOBAL_LIST_INIT(blacklisted_pylon_turfs, typecacheof(list(
...
-			if(F)
-				new /obj/effect/temp_visual/cult/turf/open/floor(F)
-			else
-				// Are we in space or something? No cult turfs or
-				// convertable turfs?
-				last_corrupt = world.time + corrupt_delay * 2
+			validturfs |= T
+
+	var/turf/T = safepick(validturfs)
+	if(T)
```
Comment (Sirryan2002) [inline code/game/gamemodes/cult/cult_structures.dm:287]: no point in null checking `T` here, this is automatically check in the istype's

File: code/game/gamemodes/cult/cult_structures.dm [hunk 8]
```diff
@@ -257,52 +268,36 @@ GLOBAL_LIST_INIT(blacklisted_pylon_turfs, typecacheof(list(
...
+		if(istype(T, /turf/simulated/floor))
+			T.ChangeTurf(/turf/simulated/floor/engine/cult)
+		if(istype(T, /turf/simulated/wall))
+			T.ChangeTurf(/turf/simulated/wall/cult/artificer)
+	else
+		var/turf/simulated/floor/engine/cult/F = safepick(cultturfs)
+		if(F)
```
Comment (Sirryan2002) [inline code/game/gamemodes/cult/cult_structures.dm:294]:
```suggestion
		if(istype(F))
```
:)

File: code/modules/tgui/tgui_datum.dm [hunk 9]
```diff
@@ -251,7 +251,7 @@
  * Run an update cycle for this UI. Called internally by SStgui
  * every second or so.
  */
-/datum/tgui/process(force = FALSE)
+/datum/tgui/process(seconds_per_tick, force = FALSE)
```
Comment (Sirryan2002) [inline code/modules/tgui/tgui_datum.dm:254]: is there a good reason behind adding a whole new param to this proc?

**Comment by ErdGinalD** [comment]
Frozen a little due to IRL problems

**Comment by Sirryan2002** [comment]
@ErdGinalD updates?

**Pull request closed without merging**